src/mlpack/methods/neighbor_search/kfn_main.cpp

```diff
@@ -261,15 +261,23 @@ static void mlpackMain()
           << queryData.n_rows << "x" << queryData.n_cols << ")." << endl;
     }
 
-    // Sanity check on k value: must be greater than 0, must be less than the
-    // number of reference points.  Since it is unsigned, we only test the upper
-    // bound.
+    // Sanity check on k value: must be greater than 0, must be less than or
+    // equal to the number of reference points.  Since it is unsigned, 
+    // we only test the upper bound.
     if (k > kfn->Dataset().n_cols)
     {
       Log::Fatal << "Invalid k: " << k << "; must be greater than 0 and less "
           << "than or equal to the number of reference points ("
           << kfn->Dataset().n_cols << ")." << endl;
     }
+    // Sanity check on k value: must not be equal to the number of reference points
+    // when query data has not been provided.
+    if (!CLI::HasParam("query") && k == kfn->Dataset().n_cols)
+    {
+      Log::Fatal << "Invalid k: " << k << "; must not be equal to the number of "
+          << "reference points (" << kfn->Dataset().n_cols << ") "
+          << "if query data has not been provided." << endl;
```

Comment:
I'm going to be a bit picky about the error message here if that's ok... this message tells the user what k cannot be but doesn't guide them for how to easily fix it.  So can I ask you to change this to something like "; must be less than the number of reference points (" please?

Reply:
Yes, that's much more appropriate. I will do it.